exoscale/api/v2.py

```diff
@@ -155,6 +156,17 @@ def _call_operation(self, operation_id, parameters=None, body=None):
         response = self.session.request(
             method=op["verb"].upper(), url=url, params=query_params, **json
         )
+
+        # Error handling
+        if response.status_code >= 400 and response.status_code < 500:
```

Comment:
A separate Exception class for auth-related errors would be useful. It can be raised whenever the API responds with a 403.